assets/src/common/components/higher-order/with-featured-image-notice.js

```diff
@@ -23,6 +24,10 @@ import { validateFeaturedImage, getMinimumFeaturedImageDimensions } from '../../
  */
 export default createHigherOrderComponent(
 	( PostFeaturedImage ) => {
+		if ( ! isObject( PostFeaturedImage ) ) {
```

Comment:
Since we're dealing with a HOC that expects an argument to be a React component, should we maybe check if `isValidElement()` here instead?

Reply:
[A function](https://github.com/WordPress/gutenberg/blob/d0e5b80f10a2feff001125dc5198f91583b39fdb/packages/editor/src/components/post-featured-image/index.js#L256) should always be passed here, so I've updated the check accordingly.